selfdrive/car/gm/interface.py

```diff
@@ -219,11 +219,16 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
       CarInterfaceBase.configure_torque_tune(candidate, ret.lateralTuning)
 
     elif candidate == CAR.SILVERADO:
-      ret.mass = 2200. + STD_CARGO_KG
+      ret.mass = 2500. + STD_CARGO_KG
```

Comment:
can you find the average curb weight of all the trims and update this in a new PR?

Reply:
average from as built of 4 samples I have know to be 2516

Reply:
![image](https://github.com/commaai/openpilot/assets/73084181/eb1a27f1-f9ed-4489-858c-a085708d01b6)

Reply:
GMC
![image](https://github.com/commaai/openpilot/assets/73084181/bca91134-920b-424e-8527-1746fdeda6ba)

Reply:
these are guide lines, but GM actually provide a sticker showing the actual weight of each vehicle built at door jam. I dont believe every denali has adaptive cruise and lane keep assists (I have seen ones without windshield camera). I believe manually sampling the values is more accurate on what weight we need to go by. Trucks has too many variations in length, accessories, tires/rim combo can all heavily affect the weight. Some trucks comes with carbon fiber too that dramatically reduces the weight. I have sampled the few individuals with OpenPilot running on their vehicle with the following weights:

2495
2509
2521
2539

Avg: 2516

These are mixed with 5.3, 6.2 and the 3.0 Diesel equipped vehicles, the weight is actually a lot closer to each other.